android/modules/geolocation/src/java/ti/modules/titanium/geolocation/GeolocationModule.java

```diff
@@ -762,7 +767,36 @@ public boolean getLocationServicesEnabled()
 												   @Kroll.argument(optional = true) final KrollFunction callback)
 	{
 		return KrollPromise.create((promise) -> {
-			tiLocation.forwardGeocode(address, createGeocodeResponseHandler(callback, promise));
+			final KrollDict response = new KrollDict();
+
+			response.put(TiC.EVENT_PROPERTY_SOURCE, this);
+
+			try {
+				final List<Address> addresses = geocoder.getFromLocationName(address, 1);
+
+				if (addresses.size() > 0) {
+					response.putAll(TiLocation.placeFromAddress(addresses.get(0)));
+				} else {
+
+					// Could not resolve address.
+					throw new Exception("Could not resolve address to location.");
+				}
+
+				// Success, resolve.
+				response.putCodeAndMessage(0, null);
+				promise.resolve(response);
+
+			} catch (Exception e) {
```

Comment:
**Question:** Will we end up in the `catch` block on a device that doesn't support reverse/forward lookup? Will this happen in an Android emulator without "Google APIs"?

That's fine if the above is true, but that would make this a breaking-change... which we have to eventually do anyways. I just want to make sure everyone is clear when this can happen. We might want to write a comment here to indicate the case(s) that cause this to happen.